### ISSUE TITLE
fcitx: switching to different window resets current selected input language

### ISSUE THREAD

**Issue by iDarkTemplar** [fcitx/fcitx#454]
After some updates I've noticed that when I'm switching to different window, my selected input language is reset to default: english.

OS: Gentoo Linux
Desktop: KDE5/X11
fcitx: 4.2.9.7
fcitx-qt5: 1.2.4
mozc: 2.23.2815.102
Languages: EN+RU, same language for all windows (i.e. when I switch to russian language and then switch to different window, selected language should stay russian)

My current configuration has 2 input languages: english (default) and russian. In addition to that I'm using fcitx+mozc for japanese input. After some updates I've noticed mentioned issue. I've been updating both KDE and fcitx recently, some other packages as well. Downgrading fcitx didn't fix this issue, and I didn't test downgrading KDE.

I've noticed that currently in the fcitx configuration I have first input method "keyboard: english", then goes "mozc" and after that goes "keyboard: russian". And when I switch to different window I assume that input method resets to default one ("keyboard: english"). I've tried to set option to tell input method state to all windows but it didn't fix the issue.

Before updates I had different configuration of input methods: "keyboard" IIRC or something similarly named and "mozc". And since it was generic keyboard, when input method was reset to default one ("keyboard"), it didn't reset my currently selected language (be it english or russian). But I can't find a generic "keyboard" input method anymore in the settings.

Basically, issue is following: if I switch to russian language (non-default), then switch to different window, input language is reset to english (default). Am I missing some configuration settings or is it a bug?

**Comment by AnezeR** [comment]
Same issue, please help

**Comment by AnezeR** [comment]
Kind of solved. Created this simple c++ script to open and close fcitx by calling the same program, then I put it to the keybind of ctrl+space, set the only fcitx option to mozc and disabled fcitx autostart. Now I get the desired outcome.
The code:
```
#include <fstream>

int main () {
        std::ifstream get_open("is_fcitx_open");
        int a;
        get_open >> a;
        if (a == 1) {
                system("killall fcitx");
                system("setxkbmap -model pc105 -layout us,ru -option  grp:alt_shift_toggle");
                get_open.close();
                std::ofstream set_open("is_fcitx_open");
                set_open << 0;
        } else {
                system("fcitx -d");
                get_open.close();
                std::ofstream set_open("is_fcitx_open");
                set_open << 1;
        }
        return 0;
}
```
